src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -89,14 +78,36 @@ public boolean remove(T object) {
         for (int i = 0; i < size; i++) {
             if (object == current.value
                     || object != null && object.equals(current.value)) {
-                remove(i);
+                unlink(current);
                 return true;
             }
             current = current.next;
         }
         return false;
     }
 
+    private void unlink(Node<T> node) {
+        Node<T> nextNode = node.next;
+        Node<T> previousNode = node.previous;
+        if (previousNode == null && nextNode != null) {
+            firstNode = nextNode;
+            nextNode.previous = null;
+        }
+        if (nextNode == null && previousNode != null) {
+            lastNode = previousNode;
+            previousNode.next = null;
+        }
+        if (nextNode != null && previousNode != null) {
+            previousNode.next = nextNode;
+            nextNode.previous = previousNode;
+        }
+        if (nextNode == null && previousNode == null) {
```

Comment:
You can make this first check then you don't need to check 
&& nextNode != null
&& previousNode != null
(nextNode != null && previousNode != null)  change to else

Reply:
fixed